File: Daily Test/Python/Minimum Time To Reach Nth Room 2.6.2025.py
n = int(input())
d = list(map(int, input().split()))
p = t = f = 0
while p < n:
    cur = d[p] ^ f
    if cur == 0:
        while p < n and (d[p] ^ f) == 0:
            p += 1
    else:
        f ^= 1
        t += 1
print(t)

File: Daily Test/Java/Minimum Time To Reach Nth Room 2.6.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] d = new int[n];
        for (int i = 0; i < n; i++) {
            d[i] = sc.nextInt();
        }

        int p = 0, t = 0, f = 0;
        while (p < n) {
            int cur = d[p] ^ f;
            if (cur == 0) {
                while (p < n && (d[p] ^ f) == 0)
                    p++;
            } else {
                f ^= 1;
                t++;
            }
        }

        System.out.println(t);
    }
}